src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/prisonperson/PrisonPersonIntTest.kt

```diff
@@ -78,19 +77,45 @@ class PrisonPersonIntTest(
 
     @Nested
     inner class Failures {
-      @Disabled("cannot implement this until we add message handling (source system is contained in the message)")
       @Test
-      fun `should ignore if source system is not DPS`() {}
+      fun `should ignore if source system is not DPS`() {
+        prisonPersonDpsApi.stubGetPrisonPerson(prisonerNumber = "A1234AA")
+        prisonPersonNomisApi.stubPutPhysicalAttributes(offenderNo = "A1234AA")
+
+        awsSnsClient.publish(physicalAttributesRequest(source = "NOMIS")).get()
+          .also { waitForAnyProcessingToComplete() }
+
+        // should never call out
+        prisonPersonDpsApi.verify(0, getRequestedFor(urlPathEqualTo("/prisoners/A1234AA")))
+        prisonPersonNomisApi.verify(0, putRequestedFor(urlPathEqualTo("/prisoners/A1234AA/physical-attributes")))
+        // should publish ignore event
+        verify(telemetryClient).trackEvent(
+          eq("physical-attributes-update-ignored"),
+          check {
+            assertThat(it).containsExactlyInAnyOrderEntriesOf(mapOf("offenderNo" to "A1234AA"))
+          },
+          isNull(),
+        )
+      }
 
       @Test
-      fun `should throw if call to DPS fails`() = runTest {
+      fun `should fail if call to DPS fails`() = runTest {
```

Comment:
possibly not worth having these tests going forward - I think we know exceptions that get thrown are retried etc etc

Reply:
Personally, I think one negative test for each external API call is reasonable. It proves that the client is wired up correctly for error handling (as appose to exceptions get retried - agree one of those is probably enough).